### DISCUSSION TITLE
feature request: search and replace

### DISCUSSION THREAD

**Discussion by psivesely** [standardnotes/forum#2498] [Suggestions] [answered]

**Comment by atmoio** [comment]
SN will resist being a text editor for as long as possible ;)

**Comment by tomcat86** [comment]
Search and highlight would be really good though. Global Search instead of per-note-window-search is OK, but highlighting the search keywords in the note-window would be extremely useful.

**Comment by BobTB** [comment]
highlighting the search FTW

**Comment by jmason888** [comment]
Especially in the code or vim editors, some quick searches would be VERY useful.  The vim editor even understands what I want to do with :s/foo/FOO/gi but refuses, saying that it "Requires searchcursor.js or any other getSearchCursor implementation"...

**Comment by atmoio** [comment]
@jmason888 I'm curious, how, and for what, are you using the Vim/Code editor day-to-day?

**Comment by M-Pixel** [comment]
> SN will resist being a text editor for as long as possible

Do you mean code editor? It does edit text, so I'm curious what the actual delineation is to you.

I have certainly used find-and-replace in the past in Writer for personal notes.

Consider...

- Story draft: Character's name changes
- Meeting notes: Learned that I've been spelling that person's name wrong

**Comment by atmoio** [comment]
I meant sort of a full-fledged word processor. I realize no one's quite asking for that, but find and replace starts taking us into that territory.

**Comment by M-Pixel** [comment]
Do you have that delineation documented somewhere? That would be useful in feature conversations like this.

I would argue that find-and-replace is a good fit for Standard Notes according to the audience and purpose it purportedly intends to serve. I was able to think of a few use cases that are easy to imagine for private personal notes, and are definitely not specific to document creation.

Even Microsoft Notepad, for example, has find-and-replace.

**Comment by atmoio** [comment]
This would be the closest read: https://standardnotes.org/longevity

**Comment by scheidm** [comment]
While I understand the "longevity" argument, who is to say how long someone's notes should be? If you cannot even search through the text, every line makes it harder to find what you are looking for in the text. Admittedly, the lack of this feature alone is enough for me to consider a refund.

**Comment by pwseo** [comment]
Having read the Longevity Statement, I would argue that *search and replace* is neither complex nor just another feature. It's an essential part of a functional text editor. Now, I know SN is not a text editor per se, but it encapsulates one, even if a very simple one (essentially a `<textarea>` for the plain-text editor).

One could argue that search and replace might be difficult to implement for all the Plus editors available, but I think it is a must for the plain text one. Otherwise we're stuck copying and pasting notes to and from SN and our normal (searchable and replaceable) text editors.

Please, do give this a look :)

**Comment by myreli** [comment, marked as answer]
While still in beta the new flagship editor [Super](https://standardnotes.com/blog/introducing-super-notes-moments-and-offline-file-access) offers the ability to search and replace! 🥳 

You can use `CMD + F` for search, `CMD + H` for search and replace or press the search icon on the mobile toolbar. [Meet Super.](https://www.youtube.com/watch?v=bkp6nrGqjUM&t=370s)

<img width="1365" alt="image" src="https://user-images.githubusercontent.com/17554234/218579869-5ea9cd85-f81a-4d94-bca6-068a1e1db34f.png">
